thirdparty/scripts/build_arrow.sh

```diff
@@ -155,7 +155,9 @@ else
     echo "Commit ID mismatches."
     git fetch origin master
     git checkout $TARGET_COMMIT_ID
+    popd
```

Comment:
Can we move the popd out of the if and have it only once (also getting rid of the else clause)?

Reply:
It is not correct to move the `pushd` statement out of the `if` statement. In this case, the `git rev-parse HEAD` will get the ray commit ID.

Reply:
Yeah, you are right, thanks for pointing that out! The solution you posted looks good to me!